caddyfile/parse_test.go

```diff
@@ -597,6 +615,56 @@ func TestEnvironmentReplacement(t *testing.T) {
 	}
 }
 
+func TestClosingCurlyBrace(t *testing.T) {
+	for i, test := range []struct {
+		input     string
+		shouldErr bool
+		keys      [][]string // keys per server block, in order
+	}{
+		{`localhost {
+			tls self_signed {}
+			}`, false, [][]string{
+			{"localhost"},
+		}},
+
+		{`localhost {
+			tls self_signed {clients verify_if_given ca.crt}
+			}`, true, [][]string{}},
+
+		{`https://localhost:8888 {
+
+			header / X-Cert-ClientSubjectDn "{tls_client_s_dn}"
+	
+			tls self_signed {clients verify_if_given ca.crt}
+	
+			}`, true, [][]string{}},
+
+		{`localhost:80 {
+			log / d:\caddy\logs\default.log   "--{user}-- {request_id}"
+			}`, false, [][]string{}},
+
+		{`localhost:80 {
+			log / d:\caddy\logs\default.log   "request_id}"
```

Comment:
This test case and the one above would fail without adding the `Quoted` property to the `Token`.  I could see no other reliable way to infer a token was derived from a quoted string and not a block.